### PR TITLE
Literal fix

### PR DESCRIPTION
**Pull request by dleslie**
A slew of small fixes related to completions and eldoc.

### CODE REVIEW

File: elisp/geiser-connection.el [hunk 1]
```diff
@@ -202,7 +202,7 @@
         `((error (key . geiser-debugger))
           (output . ,answer))
       (condition-case err
-          (let ((start (string-match "((\\(?:result)?\\|error\\) " answer)))
+          (let ((start (string-match "((\\(?:result)?\\|error\\|Error\\) " answer)))
```
Comment (jaor) [inline elisp/geiser-connection.el]: hmmm, i'd rather not modify the protocol and make chicken use lowercase, but if that complicates things unnecessarily on your side, we can keep it this way or generalize to accept result/error in a case-insensitive manner... taking Error/error but only result and no ERROR, etc. feels a bit ugly...

**Comment by dleslie** [comment]
I don't think it's possible for me to change the behaviour of Chicken's error reporting, at least not short of parsing the response and reformatting it myself.

Case-insensitive sounds like a good idea to me.

File: elisp/geiser-connection.el (hunk 1 above)
Comment (dleslie) [inline elisp/geiser-connection.el]: I don't think it's possible for me to change the behaviour of Chicken's error reporting, at least not short of parsing the response and reformatting it myself.

Case-insensitive sounds like a good idea to me.

Comment (dleslie) [inline elisp/geiser-connection.el]: Turns out its not needed thanks to the other changes made.

**Comment by dleslie** [comment]
Thank-yah!

**Pull request merged**